src/main/java/com/ibm/csync/impl/Transport.java

```diff
@@ -77,17 +80,62 @@
 		this.tracer = tracer;
 
 		req = new Request.Builder()
-			.get()
-			.url(url)
-			.build();
+				.get()
+				.url(url)
+				.build();
 
 		client = new OkHttpClient.Builder()
 			.readTimeout(60, TimeUnit.SECONDS)
 			.writeTimeout(60, TimeUnit.SECONDS).build();
 
 	}
 
-	private synchronized Futur<WebSocket> connect() {
+	public synchronized Futur<Boolean> startSession(String provider, String token) {
+		String authURL = this.req.url().toString() + encodeAuthParameters(provider, token);
+		Promise<Boolean> sessionPromise = new Promise<>();
+		if(socketFutur != null) {
+			// Already logged in
+			sessionPromise.set(null, true);
+		}
+		else {
+			connect(authURL)
+					.onComplete(workers, (ex, ws) -> {
+								if (ex != null) {
+									sessionPromise.set(ex,false);
+								}
+								else {
+									sessionPromise.set(null,true);
+								}
+							}
+
+					);
+		}
+		return sessionPromise.futur;
+	}
+
+	public synchronized Futur<Boolean> endSession() {
+		if(socketFutur == null) {
+			// Already logged out
```

Comment:
A log message here indicating that the user was not logged in would be good. Its kind of confusing on whether or not we should return a success if they were never logged in in the first place, so returning a log saying something like "Unauthenticate called when no user was logged in"